### PR TITLE
Remove jib from apps-generator plugin

### PR DESCRIPTION
**Pull request by corneil**

### CODE REVIEW

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml [hunk 1]
```diff
@@ -141,46 +141,21 @@
 				{{#app.bootPluginConfiguration}}
 				<configuration>
 					{{this}}
+					{{#app.containerImage.enableMetadata}}
+					<image>
+						<name>{{app.containerImage.orgName}}/${project.artifactId}:{{app.containerImage.tag}}</name>
+					</image>
+					{{/app.containerImage.enableMetadata}}
```
Comment (onobc) [inline spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml:148]: Do we need to specify any of the options [like before](https://github.com/spring-cloud/stream-applications/pull/548/files#diff-bdd0d0a43457d0ceee46961f59123580d51d87a88e0245d09d0e99d623fe5c1cL160-L161)? Or does the boot image already set these?

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml [hunk 2]
```diff
@@ -141,46 +141,21 @@
...
-			<!--
-			 Use of properties-maven-plugin to load the encoded metadata properties created by the jib-maven-plugin
-			 Use: Use: ./mvnw clean install jib:dockerBuild or ./mvnw clean install jib:build
-			-->
+
 			<plugin>
 				<groupId>org.codehaus.mojo</groupId>
 				<artifactId>properties-maven-plugin</artifactId>
-				<version>1.0.0</version>
+				<version>1.2.1</version>
```
Comment (onobc) [inline spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml:158]: Please put each of the version updates is a separate commit. In this PR is fine, but lets keep them separate for history sake (and be sure we do not squash - but rather rebase and merge).

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/MojoHarnessTest.java [hunk 3]
```diff
@@ -99,15 +99,8 @@ public void testSomething() throws Exception {
 
 		assertThat(plugins.stream().filter(p -> p.getArtifactId().equals("spring-boot-maven-plugin")).count()).isEqualTo(1);
 		assertThat(plugins.stream().filter(p -> p.getArtifactId().equals("properties-maven-plugin")).count()).isEqualTo(1);
-		assertThat(plugins.stream().filter(p -> p.getArtifactId().equals("jib-maven-plugin")).count()).isEqualTo(1);
-
-		Plugin jibPlugin = plugins.stream().filter(p -> p.getArtifactId().equals("jib-maven-plugin")).findFirst().get();
```
Comment (onobc) [inline spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/MojoHarnessTest.java:104]: Is there any replacements we need to make here to add coverage for the spring boot generated image?

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/SpringCloudStreamAppGeneratorMojoTest.java [hunk 4]
```diff
@@ -128,13 +127,6 @@ public void testWithDisabledContainerMetadata() throws Exception {
 		// The properties-maven-plugin should not be defined if the container metadata is not enabled.
 		assertThat(plugins.stream().filter(p -> p.getArtifactId().equals("properties-maven-plugin")).count()).isEqualTo(0);
 
-		assertThat(plugins.stream().filter(p -> p.getArtifactId().equals("jib-maven-plugin")).count()).isEqualTo(1);
```
Comment (onobc) [inline spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/SpringCloudStreamAppGeneratorMojoTest.java:131]: Ditto https://github.com/spring-cloud/stream-applications/pull/548/files#r1626275129

**Review by onobc — CHANGES_REQUESTED** [review]
All in all looks good @corneil .

I have a few minor changes requested. 

We should also consider moving the version number up to either `1.1.0` or `2.0.0` to signal the breaking change of the jib removal.

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml]: Added tests

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml (hunk 2 above)
Comment (corneil) [reply spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml:154]: Done

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/MojoHarnessTest.java (hunk 3 above)
Comment (corneil) [reply spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/MojoHarnessTest.java:104]: Added

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/resources/template/app-pom.xml]: Boot Build Image requires all labels in one environment variable. The JSON breaks the parsing of this.

File: spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/SpringCloudStreamAppGeneratorMojoTest.java (hunk 4 above)
Comment (corneil) [reply spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/test/java/org/springframework/cloud/dataflow/app/plugin/SpringCloudStreamAppGeneratorMojoTest.java:131]: Done

**Comment by corneil** [comment]
Note: The `apps-plugin-pr` workflow build successfully. Even though `ci-pr` was triggered as well because there are file changes outside `spring-cloud-dataflow-apps-plugin`

**Pull request closed without merging**